test/unit/lib/createCustomFetchSpec.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const test = require('ava');
+const http = require('http');
+const createCustomFetch = require('../../../lib/createCustomFetch');
+
+test('verify auth header', async (t) => {
+    const port = 3000;
```

Comment:
Starting a server on a hardcoded port can lead to problems when you want to run the same test in parallel (e.g. on a CI which runs for several branches at the same time). I would suggest to use something like [`test-listen`](https://www.npmjs.com/package/test-listen) instead.

Reply:
👍 Good point, added it together with the macro for testing @lo1tuma